src/pages/dar_collection_review/DarCollectionReview.js

```diff
@@ -62,33 +64,36 @@ export default function DarCollectionReview(props) {
     const init = async () => {
       //NOTE: backend is a bit restrictive, collections by id are currently not visible to anyone other than the creator
       //Endpoint needs to be adjusted, for now just remove the check on a local instance
-      const [collection, user] = await Promise.all([
-        Collections.getCollectionById(collectionId),
-        User.getMe()
-      ]);
-      const {dars, datasets} = collection;
-      const darInfo = find((d) => !isEmpty(d.data))(collection.dars).data;
-      const researcherProfile =  await User.getById(collection.createUserId);
-      const researcherProperties = {};
-      researcherProfile.researcherProperties.forEach((property) => {
-        const { propertyKey, propertyValue } = property;
-        researcherProperties[propertyKey] = propertyValue;
-      });
-      const processedBuckets = await flow([
-        generatePreProcessedBucketData,
-        processDataUseBuckets
-      ])({ dars, datasets });
-      setResearcherProperties(researcherProperties);
-      setDataUseBuckets(processedBuckets);
-      setCollection(collection);
-      setCurrentUser(user);
-      setDarInfo(darInfo);
-      setResearcherProfile(researcherProfile);
-      //setTimeout used to render skeleton loader while sub-components are initializing data for render
-      const timeout = setTimeout(() => {
-        setIsLoading(false);
-        clearTimeout(timeout);
-      }, 500);
+      const user = Storage.getCurrentUser();
+      try {
+        const collection = await Collections.getCollectionById(collectionId);
+        const { dars, datasets } = collection;
+        const darInfo = find((d) => !isEmpty(d.data))(collection.dars).data;
+        const researcherProfile = await User.getById(collection.createUserId);
+        const researcherProperties = {};
+        researcherProfile.researcherProperties.forEach((property) => {
+          const { propertyKey, propertyValue } = property;
+          researcherProperties[propertyKey] = propertyValue;
+        });
+        const processedBuckets = await flow([
+          generatePreProcessedBucketData,
+          processDataUseBuckets,
+        ])({ dars, datasets });
+        setResearcherProperties(researcherProperties);
+        setDataUseBuckets(processedBuckets);
+        setCollection(collection);
+        setCurrentUser(user);
+        setDarInfo(darInfo);
+        setResearcherProfile(researcherProfile);
+        //setTimeout used to render skeleton loader while sub-components are initializing data for render
+        const timeout = setTimeout(() => {
+          setIsLoading(false);
+          clearTimeout(timeout);
+        }, 500);
+      } catch(error) {
+        Notifications.showError({text: "Error fetching DAR collection, redirected you to console page"});
+        Navigation.console(user, props.history);
+      }
```

Comment:
Technically, this satisfies the underlying problem we've seen. The issue I see is that the only error message this is relevant to is the error from line 69. There's a lot of extra code that could, in theory, throw and error (especially the get user by id call) and in that case, the message doesn't make any sense, nor does the redirection. Not sure if it makes sense to break out the try/catch around just the collection call. Either way, I'm OK with this as a fix.

Reply:
I can change the message to "Error initializing DAR Collection page...." to make it more of a catch all (since any issues getting/setting data here means there's something wrong with the Collection and/or User responses).